be/test/exec/pipeline/pipeline_control_flow_test.cpp

```diff
@@ -1,28 +1,74 @@
 // This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
 
+#include <mutex>
+
 #include "pipeline_test_base.h"
 #include "util/thrift_util.h"
 
+#define ASSERT_COUNTER_CHUNK_NUM(counter, expected_push_chunk_num, expected_pull_chunk_num) \
+    do {                                                                                    \
+        ASSERT_EQ(expected_push_chunk_num, counter->push_chunk_num());                      \
+        ASSERT_EQ(expected_pull_chunk_num, counter->pull_chunk_num());                      \
+    } while (false)
+
+#define ASSERT_COUNTER_CHUNK_ROW_NUM(counter, expected_push_chunk_row_num, expected_pull_chunk_row_num) \
+    do {                                                                                                \
+        ASSERT_EQ(expected_push_chunk_row_num, counter->push_chunk_row_num());                          \
+        ASSERT_EQ(expected_pull_chunk_row_num, counter->pull_chunk_row_num());                          \
+    } while (false)
+
 namespace starrocks::pipeline {
 
-struct Counter {
-    size_t pull_chunk_num;
-    size_t push_chunk_num;
+class Counter {
```

Comment:
suggest to use RuntimeProfile::Counter.

Reply:
RuntimeProfile::Counter can not be reused directly